### PR TITLE
integrate embedded map and check whether guesses are correct closes #30

### LINKED ISSUE #30: Integrate embedded map
This is the core part of Capidle, it needs to show the user satelite imagery of a city, which the user can zoom in on and interact with. 

The easiest option might be to use [LeafletJS](https://leafletjs.com).

### PR DESCRIPTION
**Pull request by k-yle**
Closes #30

---
This PR intregrates the embedded map into the Capidle mini game, and also checks if the user's guess is correct. 

The map was created using the [Leaflet](http://leafletjs.com/) library. Right now there is only one game mode, which is "New Zealand towns/cities". The aerial imagery is sourced from [Toitū Te Whenua Land Information New Zealand](https://linz.govt.nz).

With the map available, a few other tweaks were required to make the game work, albeit with an ugly UI and no hints - these are coming in future PRs.

---
To test: start the app, and visit http://localhost:1234/capidle

### CODE REVIEW

File: cypress/e2e/capidle/isGuessCorrect.cy.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/// <reference types="cypress" />
```
Comment (Destiny-02) [inline cypress/e2e/capidle/isGuessCorrect.cy.js:1]: Are you able to rename this file to `is-guess-correct` to keep it consistent with the other test files?

File: src/capidle/script/isGuessCorrect.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/**
```
Comment (Destiny-02) [inline src/capidle/script/isGuessCorrect.js:1]: Are you able to rename this file to `is-guess-correct` to keep it consistent with the other files?

File: cypress/e2e/capidle/isGuessCorrect.cy.js (hunk 1 above)
Comment (Destiny-02) [reply cypress/e2e/capidle/isGuessCorrect.cy.js:1]: Same with the JSON files in `puzzles`

File: src/capidle/script/isGuessCorrect.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * and removes special characters like macrons.
+ * @param {string} guess
+ */
+const normalizeString = (guess) =>
+  guess
+    .trim()
+    .toLowerCase()
+    .normalize("NFD")
+    .replace(/\p{Diacritic}/gu, "")
+    .replace(/  +/g, " ") // remove multiple spaces
```
Comment (Destiny-02) [inline src/capidle/script/isGuessCorrect.js:12]: I am getting a SonarLint error from this line. Do you know if there's a workaround for this? 
image

**Review by Destiny-02 — CHANGES_REQUESTED** [review]
Great job on getting the map working! Left some comments on some minor things. PS: I like the 👻

File: src/capidle/script/isGuessCorrect.js (hunk 3 above)
Comment (k-yle) [reply src/capidle/script/isGuessCorrect.js]: oops, I missed the sonarlint warning... The solution seems to be `/ {2,}/`, which means _two or more of the character before the curly braces_. regex is a very weird syntax
```suggestion
    .replace(/ {2,}/g, " ") // remove multiple spaces
```

File: cypress/e2e/capidle/isGuessCorrect.cy.js (hunk 1 above)
Comment (k-yle) [reply cypress/e2e/capidle/isGuessCorrect.cy.js]: good spotting, let's be consistent with naming

**Review by Destiny-02 — APPROVED** [review]
LGTM

**Pull request merged**